### DISCUSSION TITLE
About Heterogeneous Graph Datasets

### DISCUSSION THREAD

**Discussion by AlexKarry555** [pyg-team/pytorch_geometric#6942] [Ideas]
In heterogeneous graph datasets, there are three main models and papers used: ACM, IMDB, and DBLP. Under the PyG framework, IMDB and DBLP can be directly referenced from PyG. dataset alone, but ACM datasets cannot be directly referenced.
  Currently, PyG can only reference ACM datasets from HGB data packets, but HGB data does not have a test label, which is cumbersome for experimenting with HGB datasets.
  Can we create a separate heterogeneous graph dataset for ACM, similar to the processing method of IMDB dataset

**Comment by rusty1s** [comment]
Sure, always happy to include more datasets. Do you have interest in contributing this?

**Reply by AlexKarry555** [reply]
Hello, as far as I know, the processing methods for heterogeneous graph datasets in graph neural networks are relatively messy. Specifically, when each author trains his model, the preprocessing methods for heterogeneous graph datasets are different, which also leads to the lack of a good standard dataset for evaluating heterogeneous graph models. In his paper "Are we really making multiple progress? Revising, benchmarking, and refining heterogeneous graph neural networks," Tsinghua University in China has also improved the dataset and proposed a standard dataset based on HGB, but the published dataset does not include a validation set. However, not long ago, HGB authors posted a complete dataset including the validation set on their official website, but I made an error while running the dataset code. Pyg can directly call HGB datasets, but currently the datasets seem to reference the previous (without updating the HGB datasets) link as https://github.com/THUDM/HGB 。
  Currently, IMDB and DBLP can be called from pyg (the reference is a GNN dataset: Metapath Aggregated Graph Neural Network for Heterogeneous Graph Embedding), but there is a lack of ACM dataset. This author and the author of HAN (Heterogeneous Graph Attention Network) are a team from Beijing University of Posts and Telecommunications in China, and their data processing methods are basically the same, Therefore, I suggest supplementing the ACM dataset with reference to the way HAN handles data. The link is https://github.com/Jhy1993/HAN 。
  There is also a noteworthy model, GTN (Graph Transformer Networks). The author trained three datasets, ACM, IMDB, and DBLP. These three datasets are all referenced HAN datasets, and they extracted all three datasets. The link is https://github.com/seongjunyun/Graph_Transformer_Networks Therefore, the three data sets can be obtained simultaneously from the GTN model.

**Reply by rusty1s** [reply]
> Specifically, when each author trains his model, the preprocessing methods for heterogeneous graph datasets are different, which also leads to the lack of a good standard dataset for evaluating heterogeneous graph models.

I 100% agree, it's an absolute nightmare.

I didn't know about the updated HGB examples, this is great. I am happy to include all of them, if you have some time to help me with this, this would be awesome.
